src/uct/ib/ud/base/ud_ep.c

```diff
@@ -830,41 +956,38 @@ static void uct_ud_ep_rx_creq(uct_ud_iface_t *iface, uct_ud_neth_t *neth)
                                      UCT_UD_EP_FLAG_CREQ_RCVD |
                                      UCT_UD_EP_FLAG_CREP_RCVD,
                                      "CREQ");
-        return;
+        return ep;
     }
 
     /* CREQ must have same psn */
-    ucs_assertv_always(ep->rx.ooo_pkts.head_sn == neth->psn,
-                       "iface=%p ep=%p conn_sn=%d ep_id=%d, dest_ep_id=%d"
-                       " rx_psn=%u neth_psn=%u ep_flags=0x%x ctl_ops=0x%x"
-                       " rx_creq_count=%d",
-                       iface, ep, ep->conn_sn, ep->ep_id, ep->dest_ep_id,
-                       ep->rx.ooo_pkts.head_sn, neth->psn, ep->flags,
-                       ep->tx.pending.ops, ep->rx_creq_count);
+    if (!uct_ud_ep_is_connected(ep)) {
+        ucs_assertv_always(ep->rx.ooo_pkts.head_sn == neth->psn,
+                           "iface=%p ep=%p conn_sn=%d ep_id=%d, dest_ep_id=%d"
+                           " rx_psn=%u neth_psn=%u ep_flags=0x%x ctl_ops=0x%x"
+                           " rx_creq_count=%d",
+                           iface, ep, ep->conn_sn, ep->ep_id, ep->dest_ep_id,
+                        ep->rx.ooo_pkts.head_sn, neth->psn, ep->flags,
+                         ep->tx.pending.ops, ep->rx_creq_count);
+    }
 
     /* Schedule connection reply op */
-    UCT_UD_EP_HOOK_CALL_RX(ep, neth, sizeof(*neth) + sizeof(*ctl));
-    if (uct_ud_ep_ctl_op_check(ep, UCT_UD_EP_OP_CREQ)) {
-        uct_ud_ep_set_state(ep, UCT_UD_EP_FLAG_CREQ_NOTSENT);
-    }
-    uct_ud_ep_ctl_op_del(ep, UCT_UD_EP_OP_CREQ);
```

Comment:
why removed?

Reply:
we shouldn't delete sending CREQ - peer should know that we will send him

Reply:
ok